frontend/packages/dev-console/src/components/buildconfig/sections/__tests__/EnvironmentVariablesSection.spec.tsx

```diff
@@ -0,0 +1,171 @@
+import * as React from 'react';
+import { configure, render, waitFor } from '@testing-library/react';
+import { Formik, FormikConfig } from 'formik';
+import { Provider } from 'react-redux';
+import store from '@console/internal/redux';
+import userEvent from '../../__tests__/user-event';
+import EnvironmentVariablesSection, {
+  EnvironmentVariablesSectionFormData,
+} from '../EnvironmentVariablesSection';
+
+configure({ testIdAttribute: 'data-test' });
+
+const Wrapper: React.FC<FormikConfig<EnvironmentVariablesSectionFormData>> = ({
+  children,
+  ...formikConfig
+}) => (
+  <Provider store={store}>
+    <Formik {...formikConfig}>
+      {(formikProps) => (
+        <form onSubmit={formikProps.handleSubmit}>
+          {children}
+          <input type="submit" value="Submit" />
+        </form>
+      )}
+    </Formik>
+  </Provider>
+);
+
+describe('EnvironmentVariablesSection', () => {
+  it('should render one row of input fields to enter a environment variable', async () => {
+    const initialValues: EnvironmentVariablesSectionFormData = {
+      formData: {
+        environmentVariables: [],
+      },
+    };
+    const onSubmit = jest.fn();
+
+    const renderResult = render(
+      <Wrapper initialValues={initialValues} onSubmit={onSubmit}>
+        <EnvironmentVariablesSection namespace="my-namespace" />
+      </Wrapper>,
+    );
+
+    renderResult.getByTestId('section environment-variables');
```

Comment:
Any reason you're not using `screen.getByTestId` here?

Reply:
renderResult provides also debug and container access.